system_test/m1_facilitator/Deposit.ts

```diff
@@ -0,0 +1,246 @@
+    const originChainId = config.chains.origin.chainId;
+
+    const messageHashes: string[] = [];
+
+    const initialOriginAccountBalance: Balance = {};
+    const expectedOriginAccountBalance: Balance = {};
+    const initialAuxiliaryAccountBalance: Balance = {};
+    const finalAuxiliaryAccountBalance: Balance = {};
+
+    let testDepositorAccounts = [];
+    let totalDepositorAccounts: any[] = [];
+
+    for (let i = 0; i < iterations; i += 1) {
+      // eslint-disable-next-line no-await-in-loop
+      testDepositorAccounts = await AddressHandler.getRandomAddresses(
+        depositorCount,
+        concurrencyCount,
+      );
+
+      // eslint-disable-next-line no-await-in-loop
+      await Faucet.fundAccounts(testDepositorAccounts, originChainId);
+
+      const initialBalancePromises = testDepositorAccounts.map(
+        async (account: Account): Promise<void> => {
+          const { valueToken } = config.chains.origin;
+          const { utilityToken } = config.chains.auxiliary;
+
+          const originBalance = await AddressHandler.getTokenBalance(
+            account.address,
+            originWsEndpoint,
+            valueToken,
+          );
+          initialOriginAccountBalance[account.address] = originBalance;
+
+          const auxiliaryBalance = await AddressHandler.getTokenBalance(
+            account.address,
+            auxiliaryWsEndpoint,
+            utilityToken,
+          );
+          initialAuxiliaryAccountBalance[account.address] = auxiliaryBalance;
+        },
+      );
+      // eslint-disable-next-line no-await-in-loop
+      await Promise.all(initialBalancePromises);
+
+      const depositTransactionPromises = testDepositorAccounts.map(
+        async (account: Account): Promise<void> => {
+          const { txObject, depositAmount } = await this.createDepositTransactionObject(account);
+          if (expectedOriginAccountBalance[account.address]) {
+            expectedOriginAccountBalance[account.address] += depositAmount;
+          } else {
+            expectedOriginAccountBalance[account.address] = depositAmount;
+          }
+          const txReceipt = await txObject.send({
+            from: account.address,
+            gasPrice: '0x3B9ACA00',
+            gas: (await txObject.estimateGas({ from: account.address })),
+          });
+
+          const {
+            messageHash,
+          } = txReceipt.events.DepositIntentDeclared.returnValues;
+
+          messageHashes.push(messageHash);
+        },
+      );
+
+      // eslint-disable-next-line no-await-in-loop
+      await Promise.all(depositTransactionPromises);
+      // eslint-disable-next-line no-await-in-loop
+      await new Promise(done => setTimeout(done, pollingInterval));
+
+      totalDepositorAccounts = totalDepositorAccounts.concat(testDepositorAccounts);
+    }
+
+    // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+    await new Promise(done => setTimeout(done, timeoutInterval));
```

Comment:
How do you know that this timeout interval is sufficient?

Reply:
Alternately you could wait till the certain condition is met. Like message status on target is changed. Also, add max timeout to avoid infinite wait.